src/xrpld/app/tx/detail/Escrow.cpp

```diff
@@ -456,19 +477,9 @@ EscrowFinish::doApply()
 
     if (ctx_.view().rules().enabled(featureDepositAuth))
     {
-        // Is EscrowFinished authorized?
-        if (sled->getFlags() & lsfDepositAuth)
-        {
-            // A destination account that requires authorization has two
-            // ways to get an EscrowFinished into the account:
-            //  1. If Account == Destination, or
-            //  2. If Account is deposit preauthorized by destination.
-            if (account_ != destID)
-            {
-                if (!view().exists(keylet::depositPreauth(destID, account_)))
-                    return tecNO_PERMISSION;
-            }
-        }
+        if (auto err = credentials::verify(ctx_, account_, destID, sled);
```

Comment:
I looked at the old/new code to make sure pre-amendment is going to be same in both. I think it looks right. I understand we want to make the code efficient but IMHO this is difficult to compare. I'd rather have the old code gated if the amendment is not enabled. Otherwise execute the new code. Can move this block into verify but still have it gated.

Reply:
made verifyDepositPreauth